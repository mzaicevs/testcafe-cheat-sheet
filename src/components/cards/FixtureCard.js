import React from "react";
import { fixture } from "../../snippets";
import MultiCodeCard from "../../base_components/MultiCodeCard";
import CardLink from "../../base_components/CardLink";

const FixtureCard = () => (
  <MultiCodeCard
    cardTitle={<CardLink text={fixture.title} href={fixture.fullDocLink} />}
    codeSections={fixture.codeSections}
  />
);

export default FixtureCard;
