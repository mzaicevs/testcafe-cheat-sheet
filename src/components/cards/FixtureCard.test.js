import React from "react";
import { shallow } from "enzyme";
import FixtureCard from "./FixtureCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("FixtureCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<FixtureCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
