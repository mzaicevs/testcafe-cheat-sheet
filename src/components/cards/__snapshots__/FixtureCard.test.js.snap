// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixtureCard matches snapshot 1`] = `
<CodeCard
  cardTitle={
    <CardLink
      href="https://devexpress.github.io/testcafe/documentation/test-api/test-code-structure.html#fixtures"
      text="Fixtures"
    />
  }
  codeSections={
    Array [
      Object {
        "code": "fixture( fixtureName )
fixture \`fixtureName\`",
        "description": "TestCafe tests must be organized into categories called fixtures. They accept a String parameter",
        "language": "javascript",
      },
      Object {
        "code": "fixture.page( url )
fixture.page \`url\`",
        "description": "Specify the web page URL as a String parameter where all tests in a fixture start.",
        "language": "javascript",
      },
      Object {
        "code": "fixture.meta({ key1: 'value1', key2: 'value2', key3: 'value3' })",
        "description": "Specify additional information for tests in the form of key-value metadata and use it in reports.",
        "language": "javascript",
      },
      Object {
        "code": "fixture.before( fn(ctx) )
fixture.after( fn(ctx) )",
        "description": "Fixture hooks that execute a function before/after each test run",
        "language": "javascript",
      },
      Object {
        "code": "fixture.beforeEach( fn(t) )
fixture.afterEach( fn(t) )",
        "description": "Test hooks that execute a function before/after each test.",
        "language": "javascript",
      },
      Object {
        "code": "fixture.skip",
        "description": "Skip a feature from execution.",
        "language": "javascript",
      },
      Object {
        "code": "fixture.only",
        "description": "Only execute a specific feature.",
        "language": "javascript",
      },
    ]
  }
/>
`;
